### ISSUE TITLE
Consider something other than postcss for style-objects

### ISSUE THREAD

**Issue by Georgegriff** [thompsonsj/slate-serializers#145]
Hey, I love this library 👍 

I've noticed that this file pulls in postcss-js

https://github.com/thompsonsj/slate-serializers/blob/main/packages/utilities/src/lib/style-object.ts

This library relies on Node.JS APIs which means consumers of the libraries have to handle adding nodejs polyfills etc when using with webpack. Given this is for slate and would mostly apply to being used in browsers might it be possible to consider using some other method for converting style objects?

Example errors you see when building:

```
Can't resolve 'buffer' in ...


If you want to include a polyfill, you need to:
	- add a fallback 'resolve.fallback: { "buffer": require.resolve("buffer/") }'
	- install 'buffer'
If you don't want to include a polyfill, you can use an empty module like this:
	resolve.fallback: { "buffer": false }
resolve 'buffer' in  ....
```

**Comment by thompsonsj** [comment]
Interesting. Thanks for the feedback!

TLDR: it shouldn't be there and I'll refactor it out 👍 

## Details

I never intended for `postcss-js` to be a dependency for the serializers themselves. In fact, I actively avoided it because I want the serializers to be as lightweight as possible.

As a result, I moved the `transformStyleObjectToString` and `transformStyleStringToObject` functions (which use `postcss-js`) to a separate `utilities` package as a way of making them available without including their dependencies in the serializer packages.

It seems they are being pulled in anyway because I have used them in tests.

<img width="792" alt="Screenshot 2024-07-09 at 18 37 24" src="https://github.com/thompsonsj/slate-serializers/assets/44806974/d3c9fed9-e35e-4497-b836-df1c02174ee9">
<img width="791" alt="Screenshot 2024-07-09 at 18 37 40" src="https://github.com/thompsonsj/slate-serializers/assets/44806974/52e2cb83-827f-4d47-8e24-c2d05893ffb0">

**Comment by thompsonsj** [comment]
This is the `package.json` file generated on build for `@slate-serializers/html`.

Should be enough to refactor such that the `postcss` and `postcss-js` dependencies are not included in the build. Either move the tests that use them or configure NX to exclude these dependencies.

**Comment by thompsonsj** [comment]
@Georgegriff thanks for the feedback!

Fixed in https://github.com/thompsonsj/slate-serializers/releases/tag/v2.2.0.

Required a fairly significant refactor that was long overdue - a lot of reorganisation of code but now those `postcss` dependencies are gone!

**Comment by Georgegriff** [comment]
Thank you!
